liblouis/compileTranslationTable.c

```diff
@@ -1612,11 +1611,10 @@ compileSwap(FileInfo *nested, TranslationTableOpcode opcode,
 	} else {
 		if (!compileSwapDots(nested, &replacements, &ruleDots)) return 0;
 	}
-	if (!addRule(nested, opcode, &ruleChars, &ruleDots, 0, 0, &ruleOffset, newRule,
-				noback, nofor, table))
+	if (!addRule(nested, opcode, &ruleChars, &ruleDots, 0, 0, &ruleOffset, NULL, noback,
+				nofor, table))
 		return 0;
 	if (!addRuleName(nested, &name, ruleOffset, *table)) return 0;
-	if (newRuleOffset) *newRuleOffset = ruleOffset;
```

Comment:
What confuses me a bit is that the functionality to update `newRuleOffset` seems to be completely gone. Is this not needed?

Reply:
It is indeed not needed. The `newRuleOffset` argument of `compileRule` was not used.